tools/sqlite3_api_wrapper/sqlite3_udf_api/sqlite3_udf_wrapper.cpp

```diff
@@ -0,0 +1,42 @@
+#include "sqlite3_udf_wrapper.hpp"
+#include "cast_sqlite.hpp"
+#include <cassert>
+
+#include "duckdb/common/operator/cast_operators.hpp"
+
+scalar_function_t SQLiteUDFWrapper::CreateBinarySQLiteFunction(scalar_sqlite_udf_t sqlite_udf) {
+	scalar_function_t udf_function = [=](DataChunk &args, ExpressionState &state, Vector &result) -> void {
+		assert(args.ColumnCount() == 1);
+
+		VectorData vec_data;
+		args.data[0].Orrify(args.size(), vec_data);
+
+		vector<sqlite3_value> vec_sqlite(args.size());
+		CastSQLite::ToVector(args.data[0].GetType(), vec_data, vec_sqlite);
```

Comment:
I think it should be possible to create a single function that handles an arbitrary number of arguments, rather than needing to create different functions for 1 argument, 2 arguments, 3 arguments, etc as we did in the vectorized UDFs.

The reason we did this in the vectorized UDFs is because the template specialisation on the number of arguments allows the function call to be inlined. This is significantly faster, because we save the function call per value.

For the SQLite UDFs, we unfortunately have no choice. The format of the function (a function pointer) does not allow us to inline it, and hence specialising on the number of arguments does not get us the same performance boost. 

Instead, you could create a single function that handles all the argument types here, which would perform a loop over the input vectors (`args.ColumnCount()` and constructs the `sqlite3_value` arguments for each of the function invocations.